### PR TITLE
Kdesktop 1133 unwanted popup during uninstallation on mac os

### PR DESCRIPTION
**Pull request by ChristopheLarchier**
Implemented solution: ask the client app to quit before it was killed by Sparkle to be restarted.
(Otherwise, when the server app is killed, the client app restarts it)

### CODE REVIEW

File: src/server/appserver.cpp [hunk 1]
```diff
@@ -1992,6 +1996,12 @@ void AppServer::sendErrorsCleared(int syncDbId) {
     CommServer::instance()->sendSignal(SignalNum::UTILITY_ERRORS_CLEARED, params, id);
 }
 
+void AppServer::sendQuit() {
+    int id;
```
Comment (ClementKunz) [inline src/server/appserver.cpp:2000]: Initialisation?

File: src/libcommon/utility/types.h [hunk 2]
```diff
@@ -369,7 +369,7 @@ enum class AppStateKey {
 };
 std::string toString(AppStateKey e);
 
-constexpr int64_t SELF_RESTARTE_DISABLE_VALUE = -1;
+constexpr int64_t SELF_RESTARTER_DISABLE_VALUE = -1;
```
Comment (luc-guyot-infomaniak) [inline src/libcommon/utility/types.h:372]: Here is an opportunity to rename this variable with a cameCase name.

Comment (luc-guyot-infomaniak) [inline src/libcommon/utility/types.h:372]:
```suggestion
static constexpr int64_t SELF_RESTARTER_DISABLE_VALUE = -1;
```
You may check the second answer of https://stackoverflow.com/questions/13865842/does-static-constexpr-variable-inside-a-function-make-sense regarding the use if `static` in this context.

File: src/server/updater/sparkleupdater_mac.mm [hunk 3]
```diff
@@ -43,16 +47,39 @@ - (instancetype)init {
...
+    return YES;
+}
+
+- (void)updaterWillRelaunchApplication:(SPUUpdater *)updater {
+    (void)updater;
+    LOG_DEBUG(KDC::Log::instance()->getLogger(), "Updater will relaunch app");
+
+    if (bool found = false; !KDC::ParmsDb::instance()->updateAppState(KDC::AppStateKey::LastServerSelfRestartDate,
+                                                                      KDC::SELF_RESTARTER_DISABLE_VALUE, found) ||
+                            !found) {  // Desactivate the selfRestarter
```
Comment (luc-guyot-infomaniak) [inline src/server/updater/sparkleupdater_mac.mm:73]:
```suggestion
                            !found) {  // Deactivate the selfRestarter
```

File: src/server/appserver.cpp [hunk 4]
```diff
@@ -269,6 +269,12 @@ AppServer::AppServer(int &argc, char **argv)
 #endif
     if (KDC::isVfsPluginAvailable(VirtualFileMode::Suffix, error)) LOG_INFO(_logger, "VFS suffix plugin is available");
 
+#ifdef Q_OS_MACOS
+    // Init Updater
+    std::function<void()> quitCallback = std::bind_front(&AppServer::sendQuit, this);
```
Comment (luc-guyot-infomaniak) [inline src/server/appserver.cpp:274]:
```suggestion
    const std::function<void()> quitCallback = std::bind_front(&AppServer::sendQuit, this);
```

Comment (luc-guyot-infomaniak) [inline src/server/appserver.cpp:274]: Is it possible to reuse here the `QuitCallback` type you have created below?

**Review by luc-guyot-infomaniak — COMMENTED** [review]
It would be nice if you could summarize in this PR description the most significant changes.

Are there any unit tests that could be easily added? If not, which refactoring of `appserver.cpp` would render the task of unit testing feasible? Jira ticket about it?

File: src/server/appserver.cpp (hunk 1 above)
Comment (ChristopheLarchier) [reply src/server/appserver.cpp]: Done

File: src/libcommon/utility/types.h (hunk 2 above)
Comment (ChristopheLarchier) [reply src/libcommon/utility/types.h]: Done

Comment (ChristopheLarchier) [reply src/libcommon/utility/types.h]: Done

File: src/server/updater/sparkleupdater_mac.mm (hunk 3 above)
Comment (ChristopheLarchier) [reply src/server/updater/sparkleupdater_mac.mm]: Done

File: src/server/appserver.cpp (hunk 4 above)
Comment (ChristopheLarchier) [reply src/server/appserver.cpp]: Done

Comment (ChristopheLarchier) [reply src/server/appserver.cpp]: Done

**Comment by ChristopheLarchier** [comment]
> It would be nice if you could summarize in this PR description the most significant changes.
> 
> Are there any unit tests that could be easily added? If not, which refactoring of `appserver.cpp` would render the task of unit testing feasible? Jira ticket about it?

Description updated.
There is currently no unit test for C/S functions.
https://infomaniak.atlassian.net/browse/KDESKTOP-1187?atlOrigin=eyJpIjoiYjQyZWViZTk2MWU4NDdjYmEzZjQwMjQ3ODAwNmZhYzMiLCJwIjoiaiJ9

**Review by ClementKunz — APPROVED** [review]

**Pull request merged**